iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/AbstractRelationBuilderModule.java

```diff
@@ -0,0 +1,113 @@
+     * 
+     * @param source          relation source
+     * @param target          relation target
+     * @param relType         relation type
+     * @param subRelType      relation sub-type
+     * @param relClass        relation class
+     * @param confidenceLevel an input for trust level calculation, trust level set
+     *                        to {@link StaticConfigurationProvider#ACTION_TRUST_0_9}
+     *                        when confidence level is null
+     * @param properties      relation properties
+     * @throws TrustLevelThresholdExceededException when trust level threshold exceeded
+     */
+    protected Relation createRelation(String source, String target, String relType, String subRelType, String relClass,
+            Float confidenceLevel, List<KeyValue> properties) throws TrustLevelThresholdExceededException {
+        DataInfo dataInfo = confidenceLevel != null ? buildInference(confidenceLevel)
+                : buildInferenceForTrustLevel(StaticConfigurationProvider.ACTION_TRUST_0_9);
+        return BuilderModuleHelper.createRelation(source, target, relType, subRelType, relClass,
+                properties, dataInfo, collectedFromValue);
+    }
+    
+    /**
+     * Creates an {@link AtomicAction} with {@link Relation} payload.
+     * @param source relation source
+     * @param target relation target
+     * @param relType relation type
+     * @param subRelType relation sub-type
+     * @param relClass relation class
+     * @param confidenceLevel confidence level to be used when calculating trust level
+     * @throws TrustLevelThresholdExceededException when trust level threshold exceeded
+     */
+    protected AtomicAction<Relation> createAtomicActionWithRelation(String source, String target, String relType,
+            String subRelType, String relClass, Float confidenceLevel) throws TrustLevelThresholdExceededException {
+        return createAtomicActionWithRelation(source, target, relType, subRelType, relClass, confidenceLevel, null);
+    }
+    
+    /**
+     * Creates an {@link AtomicAction} with {@link Relation} payload.
+     * @param source relation source
+     * @param target relation target
+     * @param relType relation type
+     * @param subRelType relation sub-type
+     * @param relClass relation class
+     * @param confidenceLevel confidence level to be used when calculating trust level
+     * @param properties relation properties
+     */
+    protected AtomicAction<Relation> createAtomicActionWithRelation(String source, String target, String relType,
+            String subRelType, String relClass, List<KeyValue> properties) {
+        AtomicAction<Relation> action = new AtomicAction<>();
+        action.setClazz(Relation.class);
+        try {
+            action.setPayload(createRelation(source, target, relType, subRelType, relClass, null, properties));
+        } catch (TrustLevelThresholdExceededException e) {
+            throw new RuntimeException(e);
```

Comment:
I decided to introduce `createRelation()` method version which does not accept `confidenceLevel` and which never throws `TrustLevelThresholdExceededException`. So this code was also simplified because there is no need to catch `TrustLevelThresholdExceededException`. I guess this is more straightforward.

Reply:
The price is a tiny bit of code duplication but I guess it's ok in an exceptional situation.